src/lib/party.ts

```diff
@@ -516,8 +575,105 @@ class Party {
     return new (Party as any)(body, authorization, party.isSelf);
   }
 
+  token(tokenId?: Field) {
+    let thisParty = this;
+    let customToken = new Token({
+      tokenOwner: thisParty.body.publicKey,
+      parentTokenId: tokenId,
+    });
+    return {
+      id: customToken.id,
+      parentTokenId: customToken.parentTokenId,
+      tokenOwner: customToken.tokenOwner,
+
+      mint({ address, amount }: MintOrBurnParams) {
+        let receiverParty = Party.createUnsigned(address, {
+          caller: this.id,
+          tokenId: this.id,
+          callDepth: 1,
+          useFullCommitment: Bool(true),
+        });
+
+        // Add the amount to mint to the receiver's account
+        receiverParty.body.balanceChange =
+          receiverParty.body.balanceChange.add(amount);
+      },
+
+      burn({ address, amount }: MintOrBurnParams) {
+        let receiverParty = Party.createUnsigned(address, {
+          caller: this.id,
+          tokenId: this.id,
+          callDepth: 1,
+          useFullCommitment: Bool(true),
+        });
+
+        // Sub the amount to burn from the receiver's account
+        receiverParty.body.balanceChange =
+          receiverParty.body.balanceChange.sub(amount);
+
+        // Require signature from the receiver account being deducted
+        receiverParty.authorization = {
+          kind: 'lazy-signature',
+        };
+      },
+
+      send({ from, to, amount }: SendParams) {
+        // Check if the sender party is the current zkApp address
+        if (from === thisParty.publicKey) {
+          // If so, create a new party for the zkApp to send the amount to the receiver
+          let senderParty = Party.createUnsigned(thisParty.publicKey, {
+            caller: this.id,
+            tokenId: this.id,
+          });
+          senderParty.body.balanceChange =
+            senderParty.body.balanceChange.sub(amount);
+        } else {
+          // If not, create a new party for the sender to send the amount to the receiver
+          let senderParty = Party.createUnsigned(from, {
+            caller: this.id,
+            tokenId: this.id,
+            callDepth: 1,
+            useFullCommitment: Bool(true),
+          });
+          senderParty.body.balanceChange =
+            senderParty.body.balanceChange.sub(amount);
+
+          // Require signature if the sender party is not the zkApp
+          senderParty.authorization = {
+            kind: 'lazy-signature',
+          };
+        }
+
+        let receiverParty = Party.createUnsigned(to, {
+          caller: this.id,
+          tokenId: this.id,
+          callDepth: 1,
+          useFullCommitment: Bool(true),
```

Comment:
this shouldn't set `useFullCommitment`, the receiver doesn't need replay protection